### DISCUSSION TITLE
Error when run Kong database with docker container

### DISCUSSION THREAD

**Discussion by gomslhi** [Kong/kong#11984] [Help]
I'm just learning to use Kong and I just created containers for the postgres database and when I create Kong database I got an error like this: 
>ERROR: /usr/local/share/lua/5.1/kong/tools/dns.lua:46: /etc/hosts: Permission denied
>stack traceback:
>       /usr/local/share/lua/5.1/kong/tools/dns.lua:46: in function </usr/local/share/lua/5.1/kong/tools/dns.lua:14>
>      /usr/local/share/lua/5.1/kong/globalpatches.lua:613: in function </usr/local/share/lua/5.1/kong/globalpatches.lua:12>
>        /usr/local/bin/kong:6: in function 'file_gen'
>        init_worker_by_lua(nginx.conf:116):47: in function <init_worker_by_lua(nginx.conf:116):45>
>        [C]: in function 'xpcall'
>        init_worker_by_lua(nginx.conf:116):54: in function <init_worker_by_lua(nginx.conf:116):52>

I ran it using this:
` docker run --rm --network=kong-network  -e "KONG_DATABASE=postgres"  -e "KONG_PG_HOST=kong-database"  -e "KONG_PG_PASSWORD=test"  -e "KONG_PASSWORD=test" kong/kong-gateway:3.5.0.0 kong migrations bootstrap`

I use Docker Compose version v2.21 and OS CentOS Stream 9

**Comment by StarlightIbuki** [comment]
I cannot reproduce. Could you share more potentially related details?

**Reply by gomslhi** [reply]
What kind of detailed examples can I provide?

**Reply by StarlightIbuki** [reply]
I don't know... Anything you think is related? And can you still reproduce this?

**Reply by Water-Melon** [reply]
I suspect it's a permissions issue with `/etc/hosts`. Can you check the permissions of the `hosts` file on the host machine and the container?

**Reply by gomslhi** [reply]
inside the container, the permissions for the hosts file are -rw-r-----+ and the same for the resolv.conf file, I don't know why that would be different from the permissions on the host machine, and I can't add permissions r for other users on the file.

**Comment by chronolaw** [comment]
Hi @gomslhi , could you use the latest kong 3.x and try it again?
